File: Data/Models/Company.cs
using System.ComponentModel.DataAnnotations;

namespace Nebula.Data.Models
{
    public class Company
    {
        public int Id { get; set; }
        [MaxLength(250)] public string Ruc { get; set; }
        [MaxLength(250)] public string RznSocial { get; set; }
        [MaxLength(250)] public string CodLocalEmisor { get; set; }
        [MaxLength(250)] public string TipMoneda { get; set; }
        public decimal? PorcentajeIgv { get; set; }
        public decimal? ValorImpuestoBolsa { get; set; }
        public decimal? CompletarDatosBoleta { get; set; }
        [MaxLength(250)] public string CuentaBancoDetraccion { get; set; }
        [MaxLength(250)] public string TextoDetraccion { get; set; }
        public decimal? MontoDetraccion { get; set; }
        public int? ContactId { get; set; }
    }
}

File: src/angular/src/app/account/interfaces/company.ts
export class ClaveSolHub {
  user: string = "";
  password: string = "";
}

export class Company {
  id: any = undefined;
  userId: string = "";
  ruc: string = "";
  rznSocial: string = "";
  address: string = "";
  phoneNumber: string = "";
  anchoTicket: string = "";
  codLocalEmisor: string = "";
  tipMoneda: string = "";
  porcentajeIgv: number = 0;
  valorImpuestoBolsa: number = 0;
  contactId: string = "";
  diasPlazo: number = 0;
  ubigueo: string = "";
  departamento: string = "";
  provincia: string = "";
  distrito: string = "";
  urbanizacion: string = "";
  fechaVencimientoCert: string = "-";
  sunatEndpoint: string = "FE_BETA";
  claveSol: ClaveSolHub = new ClaveSolHub();

  public getCompanyTitle(): string {
    return `${this.ruc} - ${this.rznSocial}`;
  }
}


export class CompanyInfo {
  public id: any = undefined;
  public ruc: string = "";
  public rznSocial: string = "";
}

